src/components/productList/ProductList.component.tsx

```diff
@@ -14,79 +14,72 @@ import styles from './ProductList.module.css';
 
 const ProductsList: React.FC = () => {
     const [currentPage, setCurrentPage] = useState(1);
-    const [filteredProducts, setFilteredProducts] = useState<Product[]>([]);
+    const [products, setProducts] = useState<Product[]>([]);
+    const [categories, setCategories] = useState<Category[]>([]);
     const [selectedCategories, setSelectedCategories] = useState<Category[]>([]);
     const [sortOption, setSortOption] = useState<SortOption>(SortOption.PRICE_HIGH_TO_LOW);
     const [searchQuery, setSearchQuery] = useState<string>('');
     const [searchInput, setSearchInput] = useState<string>('');
+    const [totalPages, setTotalPages] = useState(0);
+    const [isLoading, setLoading] = useState<boolean>(false);
 
     const productsPerPage = 8;
 
-    useEffect(() => {
-        let updatedProducts = mockProducts;
-
-        const filterProducts = () => {
-            if (selectedCategories.length > 0) {
-                updatedProducts = updatedProducts.filter((product) => selectedCategories.includes(product.category.name as Category));
-            }
+    const capitalizeFirstLetter = (input: string): string => input.replaceAll(/\b\w/g, (char) => char.toUpperCase());
 
-            if (searchQuery) {
-                updatedProducts = updatedProducts.filter((product) => product.title.toLowerCase().includes(searchQuery.toLowerCase()));
+    useEffect(() => {
```

Comment:
I highly suggest using a data-fetching library like React Query as it handles different ways of data-fetching, caching, prevents possible issues and provides an easy-to-use contract. 

In this case, let's always create a hook for fetching data, which will do loading, error handling.
Something like the following
```
const { data, error, isLoading } = useFetchProducts(url)
```
Learn more about why you shouldn't fetch data inside `useEffect` 
https://leejjon.medium.com/react-typescript-i-stopped-using-the-useeffect-hook-to-fetch-data-f830c1186b71
https://ah-mahir.medium.com/stop-using-useeffect-2fb4dd3e0332